test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,321 @@
+﻿using Common;
+using System;
+using System.Threading.Tasks;
+using Windows.Storage;
+using System.IO;
+using MUXControls.TestAppUtils;
+using MUXControlsTestApp.Utilities;
+using System.Text;
+using System.Collections.Generic;
+using Windows.UI.Xaml.Markup;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+using PlatformConfiguration = Common.PlatformConfiguration;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+    public enum Theme
+    {
+        Dark,
+        Light,
+        HC,
+        All
+    }
+
+    /// <summary>
+    /// TestBaseClass helps to compare visualtree.
+    /// Don't add <c>[TestInitialize]</c> in your test case, instead, just override <see cref="OnTestInitialized"/>
+    ///  <see cref="VerifyVisualTree"/> helps to compare the visualtree with a 'master' file.
+    /// Master filename for testcase would [testclass]_[testname]_[theme].xml or [testclass]_[testname]_[theme]-[apiversion].xml
+    /// </summary>
+    public class VisualTreeTestBase
```

Comment:
I would avoid using a baseclass to enable this functionality. Inherritance is inflexible and not easily composed.

Consider instead creating a helper object that can be instantiated from any test method.

Reply:
It's used to get the TestContext, otherwise every test class would have to have its own TestInitialize. It will simply most of the testing. I didn't see any inflexible for visualtree testing yet.

---
In reply to: [243670166](https://github.com/Microsoft/microsoft-ui-xaml/pull/126#discussion_r243670166) [](ancestors = 243670166)

Reply:
+1 Please avoid the requirement of base class. what do you need from the test context? can that be passed in somehow ?

Reply:
Removed baseclass and use VisualTreeTestHelper